### PR TITLE
Support building types

### LINKED ISSUE #363: Support Standards Building Type and Standards Space Type in Space Type data
Currently other than space type name, color is only data that can be stored. It would be great to include Standards Building Type and Standards Space Type. This would make it easier to use measure to import JSON into an empty seed model and then run measures that populates space types using the standards gem.

### PR DESCRIPTION
**Pull request by benfen**
Resolves https://github.com/NREL/floorspace.js/issues/363

Adds in a select field for building types in the assignment section of the UI.  There is a default list of building types provided for the project, but the user can modify the list however they want.  There is also a freeform text input for template that has been added.  The `building_unit` field of a space is used to bind spaces into the same building.  Modifying the building type of a space that shares the same building unit as another space will change the building type of that space as well; conversely, modifying the building unit of a space will change that space's building type to match the building type of all spaces in that building unit.

In addition to this, the space types select now reacts to changes in building type and template to pre-generate a list of space types for recognized combinations of building type and template.  These space types are always appended after user-defined space types.

The list of space types was pulled from [here](https://github.com/NREL/openstudio-extension-gem/blob/develop/lib/openstudio/extension/core/os_lib_model_generation.rb#L404-L696).  I'm not sure what to do with the extra information stored there - I assume we can just drop it for now, but would appreciate clarification.  Once I have that I can clean up the `helpers.js` file all that code got dumped into.

### CODE REVIEW

File: src/store/modules/models/helpers.js [hunk 1]
```diff
@@ -195,5 +195,427 @@ const helpers = {
...
+
+  /**
+   * Generates a hash containing all the space types for the given building type
+   *
+   * @param {string} building_type Type of building to generate space types for
+   * @param {string} template Template that the space is following
+   * @param {boolean} wholeBuilding Whether or not the space types are generated for the whole building
+   * @returns {{[key: string]: string}}
+   */
+  spaceTypesForBuilding(building_type, template, wholeBuilding = false) {
```
Comment (benfen) [inline src/store/modules/models/helpers.js:209]: This logic is pulled from https://github.com/NREL/openstudio-extension-gem/blob/develop/lib/openstudio/extension/core/os_lib_model_generation.rb#L404-L696.  If I know for sure that we don't need the additional data there,  I can simplify the logic here and cut down on some of the redundancy.

File: src/store/modules/models/mutations.js [hunk 2]
```diff
@@ -38,37 +38,72 @@ export default {
...
+   * If the building unit is changed, it will change the type of the building to match all other spaces with that building unit
+   *
+   * @param {*} state State to update
+   * @param {*} payload Payload with data fields to update on the space
+   */
+  updateSpaceWithData(state, payload) {
+    const space = payload.space;
+
+    if (payload.building_type_id && space.building_unit_id) {
+      state.stories.forEach((story) => {
```
Comment (kylem038) [inline src/store/modules/models/mutations.js]: I'm not sure if a map would be more performant here per the comment you left on my last PR. https://github.com/NREL/floorspace.js/pull/399#discussion_r673392804

**Review by kylem038 — COMMENTED** [review]
Great work! Also, thank you for fixing up the inconsistent white spacing. It's been driving me nuts. Left one possible performance comment otherwise looks good to me.

File: src/store/modules/models/mutations.js (hunk 2 above)
Comment (benfen) [reply src/store/modules/models/mutations.js]: Yeah, I can change that

File: src/store/modules/models/helpers.js (hunk 1 above)
Comment (bgschiller) [reply src/store/modules/models/helpers.js:209]: it looks like the keys and values are always identical. Maybe a Set or Array would be a better choice of data structure?

File: test/unit/specs/models/mutations.spec.js [hunk 3]
```diff
@@ -1,21 +1,12 @@
...
-      "below_floor_plenum_height":0,"floor_to_ceiling_height":0,"multiplier":0,"spaces":[{"id":"13","name":"Living 3","color":"#8BC600","handle":null,"face_id":null,"daylighting_controls":[],"building_unit_id":null,"thermal_zone_id":null,"space_type_id":"17","construction_set_id":null},{"id":"16",
-      "name":"Attic 3","color":"#557a00","handle":null,"face_id":null,"daylighting_controls":[],"building_unit_id":null,"thermal_zone_id":null,"space_type_id":"21","construction_set_id":null}],"windows":[],"shading":[],"images":[],"geometry_id":"12"}],"library":{"building_units":[],"thermal_zones":[
-      {"id":"22","color":"#007373","name":"Thermal Zone 1"},{"id":"23","color":"#000000","name":"Thermal Zone 2"}],"space_types":[{"id":"17","color":"#007373","name":"Living","type":"space_types"},{"id":"18","color":"#000000","name":"Basement","type":"space_types"},
-      {"id":"19","color":"#004749","name":"Crawlspace","type":"space_types"},{"id":"20","color":"#009196","name":"Garage","type":"space_types"},{"id":"21","color":"#8BC600","name":"Attic","type":"space_types"}],"construction_sets":[],"windows":[],"daylighting_controls":[]}}`);
```
Comment (bgschiller) [inline test/unit/specs/models/mutations.spec.js:17]: :fire:

**Review by bgschiller — APPROVED** [review]

File: src/store/modules/models/helpers.js (hunk 1 above)
Comment (benfen) [reply src/store/modules/models/helpers.js:209]: The UI expects to be returned an object with id - name pairings; however, I forgot to update the names to be user friendly, though, so I'll go through and update them.

**Pull request merged**